definitions/features/package_manager.rb

```diff
@@ -28,6 +28,22 @@ def manager
                  end
   end
 
+  def satellite_installed?
+    installed?([satellite_package]) || (find_package('foreman') =~ /sat.noarch/)
```

Comment:
The satellite_package method returns 'satellite' statically, so above line will never end up using find_package method ?

Reply:
I am not sure from what version of satellite, it contains rpm package called `satellite`. If it is from satellite 6.2 then probably then we don't need a second condition.

Reply:
@kgaikwad I don't think PackageManager is proper place for these methods. The class should be generic tool for package manipulation and we should avoid putting inside methods with specific packages.

Would it be possible to make feature(:satellite) the single point of truth and doing the check in the confine?
Then in the feature(:instance) we could have `confine do feature(:satellite) || feature(:capsule) end`
Wouldn't it make better sense? Similar for capsule.

Reply:
Yes, makes sense! :+1:

Reply:
>Then in the feature(:instance) we could have confine do feature(:satellite) || feature(:capsule) end

Should I add feature(:katello) as well in confine condition? I think few users still use foreman_maintain for `backup`?

Reply:
Sorry, now I remember why I have moved these methods i.e `satellite_installed? as well as capsule_installed?` to feature package_manager. The reason behind this is that I need to check `feature(:instance).downstream?` in the confine of feature capsule.

Reply:
IMO `feature(:capsule)` should be lower level then the `:instance.downstream?` which is to me rather helper. So `:capsule` should do the work and `downstream?` should just check for existence of `:satellite` or `:capsule`. Is there any use-case where this won't work?

Reply:
@kgaikwad I don't get the question about `:katello` and backup. That may be due to my wrong comment - by `confine do feature(:satellite) || feature(:capsule) end` I meant `def downstream?  feature(:satellite) || feature(:capsule) end` - was that it? Sorry for the confusion. `feature(:instance)` is rather helper feature and should be present always with no confine block.

Reply:
Yes, got it.